### DISCUSSION TITLE
Would be useful in editor to be able to reposition parent nodes without affecting global transform of children

### DISCUSSION THREAD

**Discussion by blackears** [godotengine/godot-proposals#6511] [2D]
I've been working on a skeleton for a 2D character and am finding it difficult to position bone joints where I need them since moving a parent bone will move all of its children too.  It would be nice if you could have a flag added to the move tool so that when you move a node, it automatically updates the transforms of all its children so that they remain in the same position relative to world space.

Programs like Blender have a special edit mode for this so that you can reposition the origin of a bone while not moving any of the children.

**Comment by ka0s420** [comment]
with parent selected and move tool selected look at tool options along top and you will notice a padlock symbol, showing locked icon, click it to unlock and then select child node and move it, re-select parent node and click padlock icon again to relock and voila.

**Reply by jeanmarc** [reply]
OP means to be able to do this the other way around: lock the children so they stay fixed, and then reposition the parent.

**Comment by tomsterBG** [comment]
Haven't seen it implemented so i'll mention, it may be useful to consider whether the control scheme will be toggle or hold to enable. For example like how Ctrl-moving makes moving snapped to grid only wwhen holding Ctrl. Ideally both options should be available, let us toggle it on and off and have a key to bind for this "hold to move without affecting children" mode.

**Comment by ryevdokimov** [comment]
Made a PR, would appreciate feedback:

https://github.com/godotengine/godot/pull/109680
